### PR TITLE
Add support for website quirks

### PR DESCRIPTION
**Pull request by mre**
This is an attempt at #28. 
I've added a structure called `Quirk` and a few tests for Twitter and YouTube.

~~It's very hacky right now and definitely a work-in-progress. Early feedback welcome.
Need to do a lot of cleanup before this can be merged.~~

This is not perfect, but I'd say it's okay to merge this now.
The YouTube oembed check doesn't work somehow; I get a 404 at least.
I don't know why, but we can revisit this in the future. Maybe you have an idea @pawroman? 😉

### CODE REVIEW

File: src/quirks/mod.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            Quirk {
+                // https://stackoverflow.com/a/19377429/270334
+                pattern: Regex::new(r"^(https?://)?(www\.)?(youtube\.com|youtu\.?be)").unwrap(),
+                rewrite: |request| {
+                    let mut out = request;
+                    let original_url = out.url();
+                    let urlencoded: String =
+                        url::form_urlencoded::byte_serialize(original_url.as_str().as_bytes())
+                            .collect();
+                    *out.method_mut() = Method::HEAD;
```
Comment (pawroman) [inline src/quirks/mod.rs:41]: It seems that making a GET request works here, while HEAD does not (verified with curl).

File: src/quirks/mod.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    pub rewrite: fn(Request) -> Request,
+}
+
+#[derive(Debug, Clone)]
+pub struct Quirks {
+    quirks: Vec<Quirk>,
+}
+
+impl Quirks {
+    pub fn init() -> Self {
```
Comment (pawroman) [inline src/quirks/mod.rs:20]: Why not `new`?

If the intention was to indicate that it's going to "init standard quirks" then maybe we should be a bit more verbose with the name, e.g.

`pub fn standard_quirks() -> Self {`

It's just a nit-picking suggestion though.

**Review by pawroman — COMMENTED** [review]
Nice work, looks solid. Left a few minor comments, but otherwise all good with me!

File: src/quirks/mod.rs (hunk 2 above)
Comment (mre) [reply src/quirks/mod.rs:20]: You're right. Come to think of it, there is a default way to handle that scenario and that is... 🥁 🥁 🥁 the `Default` trait. 🎉 
So I'll implement that instead. `Quirks::init()` becomes `Quirk::default()` then, which is closer to what I'd like to convey.

File: src/quirks/mod.rs (hunk 1 above)
Comment (mre) [reply src/quirks/mod.rs]: You're right. Checked on my machine and got the same results. Fixed now and added a unit test for verification. Thanks!

**Pull request merged**